pallets/pallet-bonded-coins/src/lib.rs

```diff
@@ -728,22 +763,6 @@ pub mod pallet {
 				// if burnt <= sum_of_issuances, which should always hold true
 				.ok_or(Error::<T>::Internal)?;
 
-			if amount.is_zero()
-				|| T::CollateralCurrencies::can_deposit(
-					pool_details.collateral_id.clone(),
-					&who,
-					amount,
-					Provenance::Extant,
-				)
-				.into_result()
-				.is_err()
-			{
-				// Funds are burnt but the collateral received is not sufficient to be deposited
-				// to the account. This is tolerated as otherwise we could have edge cases where
-				// it's impossible to refund at least some accounts.
-				return Ok(());
```

Comment:
this is a bigger problem, because we have two big edge cases:
1. There can be less collateral than bonded currency, which means that some could receive 0 refunds. Either we allow reaping these accounts (the way it currently works), or you need to refund other accounts first, hope that this has brought the total supply to 0, and if not, try refunding this account again.
2. Worse: You may be entitled to so little collateral that you cannot transfer it to your account due to minimum balance requirements. In that case your account cannot be refunded and the whole pool teardown is stuck.

Reply:
I reintroduced the check: [2bced52](https://github.com/KILTprotocol/kilt-node/pull/789/commits/2bced5208c970b38a109a4c74063769d82407e16)